pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -976,9 +994,15 @@ func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeNam
 
 	if !nodeExist {
 		glog.Warningf(
-			"Node %q does not exist. DisksAreAttached will assume vmdk %v are not attached to it.",
+			"Node %q does not exist. DisksAreAttached will delete any vmdk %v attached to it.",
 			vSphereInstance,
 			volPaths)
+		for _, volPath := range volPaths {
+			if err = vs.DetachDisk(volPath, nodeName); err != nil {
+				glog.V(1).Infof("Error deleting vsphere volume %s: %v", volPath, err)
+				return attached, err
+			}
```

Comment:
Actually since this function is only for checking the status, maybe we should put detachdisk function inside of it. When node node no longer exists, we should return an error message instead of returning false indicating volume is already detached.